packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3248,30 +3249,87 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
   /// the global coordinate system in logical pixels. To get physical pixels,
   /// use [applyPaintTransform] from the [RenderView] to further transform the
   /// coordinate.
+  ///
+  /// See also:
+  ///
+  ///  * [computeTransformToRenderObject] which does the same, but the target
+  ///    `RenderObject` does not have to be an ancestor.
   Matrix4 getTransformTo(RenderObject? ancestor) {
-    final bool ancestorSpecified = ancestor != null;
     assert(attached);
-    if (ancestor == null) {
-      final RenderObject? rootNode = owner!.rootNode;
-      if (rootNode is RenderObject) {
-        ancestor = rootNode;
-      }
+    final Matrix4 identity = Matrix4.identity();
+    if (ancestor != null) {
+      return _computePaintTransform(ancestor, identity);
+    } else  {
+      final RenderObject? parent = this.parent;
+      return parent != null ? _computePaintTransformToGlobalCoordinates(parent, identity) : identity;
```

Comment:
I think when `ancestor` is null the expected behavior of the method is to find the closest `RenderView` (since  `RenderView.applyPaintTransform` applies the device's pixel ratio)?

Reply:
I would have expected that it just walks all the way to root node of the pipeline owner without making any assumptions what that root node is. As documented, if people want to convert to physical pixels, they'd have to do that manually by applying the transform of the RenderView.